### PR TITLE
updated payment verfication

### PR DESCRIPTION
**Pull request by ankitdas13**
for verify_payment_link_signature : https://github.com/razorpay/razorpay-node/blob/master/documents/paymentVerfication.md#verify-paymentlink-verification
Link : https://razorpay.com/docs/api/payments/payment-links/#verify-signature

### CODE REVIEW

File: lib/razorpay/utility.rb [hunk 1]
```diff
@@ -10,9 +10,15 @@ def self.verify_payment_signature(attributes)
...
       data = attributes.sort.map(&:last).join('|')
-
       secret = Razorpay.auth[:password]
+      verify_signature(data, signature, secret)
+    end
 
+    def self.verify_payment_link_signature(attributes)
+      signature = attributes.delete(:razorpay_signature)
+      # element of each is the value. These are joined.
+      data = attributes.map(&:last).join('|')
```
Comment (arjunmenon) [inline lib/razorpay/utility.rb:20]: payment_response.values.join("|")

File: test/razorpay/test_utility.rb [hunk 2]
```diff
@@ -21,6 +21,24 @@ def test_payment_signature_verification
...
+        razorpay_payment_id: 'pay_IH3d0ara9bSsjQ',
+        razorpay_signature: 'b8a6acda585c9b74e9da393c7354c7e685e37e69d30ae654730f042e674e0283'
+      }
+
+      Razorpay::Utility.verify_payment_link_signature(payment_response)
+
+      payment_response[:razorpay_signature] = '_dummy_signature' * 4  
+      assert_raises(SecurityError) do
+        Razorpay::Utility.verify_payment_link_signature(payment_response)
+      end
```
Comment (arjunmenon) [inline test/razorpay/test_utility.rb:39]: Do add a positive test case as well

Comment (ankitdas13) [reply test/razorpay/test_utility.rb:39]: added postive test cases

**Review by arjunmenon — APPROVED** [review]
LGTM

**Pull request merged**